src/Wrapper.js

```diff
@@ -73,15 +74,37 @@ class Wrapper extends React.Component {
     // const { appId, } = this.state.appInstance
     // this.openApp(appId, params)
   }
-  handleTransaction = ({ transaction }) => {
-    this.showWeb3ActionSigner(
-      {
-        to: transaction && transaction.to,
-        tx: transaction,
-        description: transaction && transaction.description,
-      },
-      { error: null }
-    )
+  makeTransactionIntent(transaction = {}) {
+    const { apps } = this.props
+    const { description, to } = transaction
+    const toApp = apps.filter(app => addressesEqual(app.proxyAddress, to))[0]
```

Comment:
You can use [`Array#find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) here instead